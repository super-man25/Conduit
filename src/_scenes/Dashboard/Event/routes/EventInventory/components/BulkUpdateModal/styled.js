import styled, { keyframes } from 'styled-components';
import { cssConstants } from '_constants';
import { Box, Text, H4 } from '_components';

export const slideUpAnimation = keyframes`
  0% {
    opacity: 0;
    transform: translate(-50%, -20%);
  }
  100% {
    opacity: 1;
  }
`;

export const ModalOverlay = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  opacity: 0.3;
  background-color: ${cssConstants.PRIMARY_LIGHT_BLACK};
`;

export const ModalContent = styled.div`
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 350px;
  z-index: 9;
  box-shadow: 0 1px 10px rgba(0, 0, 0, 0.27);
  animation: ${slideUpAnimation} 200ms ease-out;
  animation-fill-mode: forwards;
`;

export const ModalHeader = styled.div`
  padding: 1.5rem;
  background-color: ${cssConstants.PRIMARY_LIGHTEST_GRAY};
  border-bottom: 1px solid ${cssConstants.PRIMARY_BLUE};
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
`;

export const ModalBody = styled.div`
  padding: 1.5rem;
  background-color: ${cssConstants.PRIMARY_WHITE};
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 3px;
`;

export const Title = styled(H4)`
  margin: 0;
  margin-bottom: 0.25rem;
  color: ${cssConstants.PRIMARY_LIGHT_BLACK};
`;

export const NumberInput = styled.input`
  background-color: ${cssConstants.PRIMARY_WHITE};
  border: 1px solid ${cssConstants.PRIMARY_GRAY};
  border-radius: 3px;
  box-sizing: border-box;
  height: 40px;
  margin: 0;
  padding: 0 0.5rem;
  width: 100%;
`;

export const Label = styled.label`
  display: block;
  margin-bottom: 0.5rem;
  color: ${cssConstants.PRIMARY_LIGHT_BLACK};
`;

export const Field = styled(Box)`
  position: relative;
`;

export const FieldErrorText = styled(Text)`
  position: absolute;
  top: 100%;
`;
