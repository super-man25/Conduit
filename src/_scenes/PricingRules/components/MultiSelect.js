// @flow
import * as React from 'react';
import { Text, Flex, Icon } from '_components';
import { colors } from '_constants';
import {
  MultiSelectContainer,
  MultiSelectMenu,
  SplitButtonContainer,
  SplitButtonHalf,
  DropdownFilterInput,
  DropdownFilterInputContainer,
} from './styled';

import { MultiSelectGroupView } from './MultiSelectGroupView';
import { MultiSelectView } from './MultiSelectView';

export type Option = {
  id: number,
};

type State = {
  isOpen: boolean,
  selectAllNext: boolean,
  isGrouped: boolean,
  filterText: string,
};

type Props = {
  options: any[],
  selected: any[],
  disabled: string[],
  labelFn: (option: any) => string,
  toggleSelectAll: (selectAllNext: boolean) => void,
  label: string,
  cellLabel: string,
  gridProps: any,
  rows?: any,
  isGroupable: boolean,
  columnData: any,
  updatePriceRuleProperty: (option: any[]) => void,
  onItemClicked: (option: any) => void,
};

type GroupedSplitButtonProps = {
  isGrouped: boolean,
  selectGrouping: (grouped: boolean) => void,
};

export function GroupedSplitButton({
  isGrouped,
  selectGrouping,
}: GroupedSplitButtonProps) {
  return (
    <SplitButtonContainer>
      <SplitButtonHalf
        isActive={!isGrouped}
        onClick={() => selectGrouping(false)}
      >
        Uncategorized
      </SplitButtonHalf>
      <SplitButtonHalf
        isActive={isGrouped}
        onClick={() => selectGrouping(true)}
      >
        By Category
      </SplitButtonHalf>
    </SplitButtonContainer>
  );
}

export class MultiSelect extends React.Component<Props, State> {
  state = {
    isOpen: false,
    selectAllNext: true,
    isGrouped: false,
    filterText: '',
  };
  ref = React.createRef<HTMLDivElement>();

  componentDidMount() {
    if (this.props.options.length === this.props.selected.length) {
      this.setState({ selectAllNext: false });
    }
  }

  toggleDropdownOpen = () => {
    this.setState({ isOpen: !this.state.isOpen });
  };

  openDropdown = () => {
    if (this.state.isOpen) return;
    this.setState({ isOpen: true });
  };

  getMultiSelectHeight = () => {
    if (!this.ref.current) return {};

    const { height, headerHeight } = this.props.gridProps;
    const { bottom } = this.ref.current.getBoundingClientRect();

    const multiSelectHeight = height + headerHeight - bottom;

    return `${multiSelectHeight}px`;
  };

  toggleSelectAll = () => {
    const { toggleSelectAll } = this.props;
    const { selectAllNext } = this.state;

    toggleSelectAll(selectAllNext);
    this.setState({ selectAllNext: !selectAllNext });
  };

  selectGrouping = (isGrouped: boolean) => {
    this.setState({ isGrouped });
  };

  render() {
    const { isOpen, selectAllNext, filterText } = this.state;
    const {
      cellLabel,
      label,
      selected,
      disabled,
      options,
      isGroupable,
      columnData,
      labelFn,
      onItemClicked,
      updatePriceRuleProperty,
    } = this.props;

    let OptionsView;
    if (isGroupable && this.state.isGrouped) {
      const {
        grouping: { categoriesKey, groupedKey },
      } = columnData;
      const filteredOptions = {};
      Object.keys(columnData[groupedKey]).map(
        (key) =>
          (filteredOptions[key] = columnData[groupedKey][key].filter(
            (option) => {
              return (
                option.id === filterText ||
                option.name.toLowerCase().includes(filterText.toLowerCase())
              );
            }
          ))
      );
      const filteredCategories = columnData[categoriesKey].filter(
        (category) => filteredOptions[category.id].length > 0
      );
      OptionsView = (
        <MultiSelectGroupView
          categories={filteredCategories}
          grouped={filteredOptions}
          selected={selected}
          labelFn={labelFn}
          updatePriceRuleProperty={updatePriceRuleProperty}
          onItemClicked={onItemClicked}
        />
      );
    } else {
      const filteredOptions = filterText
        ? options.filter((option) => {
            if (columnData.optionsKey === 'buyerTypes') {
              return (
                option.id === filterText ||
                option.publicDescription.toLowerCase().includes(filterText) ||
                option.code.toLowerCase().includes(filterText.toLowerCase())
              );
            } else if (columnData.optionsKey === 'priceScales') {
              return (
                option.id === filterText ||
                option.name.toLowerCase().includes(filterText.toLowerCase())
              );
            } else if (columnData.optionsKey === 'events') {
              return (
                option.id === filterText ||
                option.name.toLowerCase().includes(filterText.toLowerCase())
              );
            }
            return true;
          })
        : options;
      OptionsView = (
        <MultiSelectView
          options={filteredOptions}
          selected={selected}
          disabled={disabled}
          labelFn={labelFn}
          onItemClicked={onItemClicked}
        />
      );
    }

    return (
      <MultiSelectContainer
        onClick={this.openDropdown}
        onClickAway={() => this.setState({ isOpen: false })}
      >
        {cellLabel}
        <Icon
          color={isOpen ? colors.blue : colors.black}
          name={'arrowDown'}
          size={12}
        />
        <div
          ref={this.ref}
          style={{
            position: 'fixed',
            zIndex: 9,
          }}
        >
          <MultiSelectMenu
            show={isOpen}
            style={{ maxHeight: this.getMultiSelectHeight() }}
          >
            <DropdownFilterInputContainer>
              <DropdownFilterInput
                value={this.state.filterText}
                onChange={({ target: { value } }) =>
                  this.setState({ filterText: value })
                }
                placeholder="Filter..."
              />
            </DropdownFilterInputContainer>
            <Flex
              align="center"
              justify="space-between"
              padding="1rem"
              flexWrap="wrap"
              style={{ backgroundColor: colors.white }}
            >
              <Text size={13} weight={600}>
                {label}
              </Text>
              <Text
                color={colors.blue}
                style={{ cursor: 'pointer' }}
                onClick={() => this.toggleSelectAll()}
              >
                {selectAllNext ? 'Select All' : 'Select None'}
              </Text>

              {isGroupable && (
                <GroupedSplitButton
                  isGrouped={this.state.isGrouped}
                  selectGrouping={this.selectGrouping.bind(this)}
                />
              )}
            </Flex>
            {OptionsView}
          </MultiSelectMenu>
        </div>
      </MultiSelectContainer>
    );
  }
}
