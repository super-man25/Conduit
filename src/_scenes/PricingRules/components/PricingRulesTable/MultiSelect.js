// @flow
import * as React from 'react';
import { Text, Flex, Box } from '_components';
import { cssConstants } from '_constants';
import {
  MultiSelectList,
  MultiSelectListItem,
  Checkbox,
  MultiSelectContainer
} from './styled';

export type Option = {
  id: number
};

type Props = {
  items: any[],
  labelFn: (item: Option) => string,
  selectedItems: any[],
  onItemClicked: (item: Option) => void,
  toggleSelectAll: () => void,
  selectAllNext: boolean,
  label: string
};

export const MultiSelect = ({
  items,
  labelFn,
  selectedItems,
  onItemClicked,
  toggleSelectAll,
  selectAllNext,
  label
}: Props) => (
  <MultiSelectContainer>
    <Flex align="center" justify="space-between" padding="1rem">
      <Text size={13} weight={600}>
        {label}
      </Text>
      <Text
        color={cssConstants.PRIMARY_LIGHT_BLUE}
        style={{ cursor: 'pointer' }}
        onClick={() => toggleSelectAll()}
      >
        {selectAllNext ? 'Select All' : 'Select None'}
      </Text>
    </Flex>
    <Box height="400px">
      <MultiSelectList>
        {items.map((item) => (
          <MultiSelectListItem key={item.id}>
            <Checkbox
              checked={selectedItems.includes(item.id)}
              onChange={() => onItemClicked(item)}
            />
            <Text size={13}>{labelFn(item)}</Text>
          </MultiSelectListItem>
        ))}
      </MultiSelectList>
    </Box>
  </MultiSelectContainer>
);
