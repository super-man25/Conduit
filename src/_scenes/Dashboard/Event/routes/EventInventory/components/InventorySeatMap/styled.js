import React from 'react';
import { Box, CenteredLoader } from '_components';
import styled from 'styled-components';

export const EventInventoryVerticalRule = styled(Box)`
  background-color: #979797;
  cursor: pointer;
`;

export const SeatIcon = styled.img`
  cursor: pointer;
`;

export const SeatMapLoader = () => (
  <div style={{ position: 'relative', height: '100%', minHeight: 400 }}>
    <CenteredLoader />
  </div>
);
