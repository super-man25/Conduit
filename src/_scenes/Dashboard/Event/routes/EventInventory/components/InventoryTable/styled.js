import styled from 'styled-components';
import { cssConstants } from '_constants';
import { Button, Input, Flex, Box } from '_components';

export const ManualPricingInput = Input.extend`
  box-sizing: border-box;
  margin: 0 0.5rem 0 0;
  padding: 10px 16px;
  max-height: 35px;
  max-width: 80px;
`;

export const ManualPricingButton = Button.extend`
  box-sizing: border-box;
  margin: 0;
  max-width: 80px;
  max-height: 35px;
`;

export const TableHeaderCell = styled.div`
  text-align: ${(props) => props.align || 'left'};
  color: ${cssConstants.PRIMARY_DARKEST_GRAY};
  font-weight: 500;
`;

export const IconContainer = Flex.extend`
  display: inline-flex;
  cursor: pointer;
  position: absolute;
  left: 0;
  top: 50%;
  transform: translateY(-50%);
  opacity: 0.667;
`;

export const PositionedBox = Box.extend`
  position: ${(props) => props.position};
  top: ${(props) => props.top};
  bottom: ${(props) => props.bottom};
  left: ${(props) => props.left};
  right: ${(props) => props.right};
  z-index: ${(props) => props.zIndex};
`;

export const ScaleFilterContainer = Box.extend`
  width: 280px;
  border: 1px solid ${cssConstants.SECONDARY_BLUE};
  background-color: ${cssConstants.PRIMARY_WHITE};
  position: relative;
  box-shadow: 0 2px 5px 0px rgba(0, 0, 0, 0.43);

  &::before {
    content: '';
    width: 8px;
    height: 8px;
    position: absolute;
    top: -5px;
    left: 43px;
    transform: rotate(45deg);
    z-index: 1;
    background-color: ${cssConstants.PRIMARY_WHITE};
    border-top: 1px solid ${cssConstants.SECONDARY_BLUE};
    border-left: 1px solid ${cssConstants.SECONDARY_BLUE};
  }
`;

export const ScaleFilterList = styled.ul`
  margin: 0;
  height: 100%;
  list-style: none;
  padding: 0;
  overflow: auto;
`;

export const ScaleFilterListItem = styled.li`
  height: 35px;
  display: flex;
  padding: 0 1rem;
  align-items: center;
`;

export const Checkbox = styled.input.attrs({ type: 'checkbox' })`
  margin-right: 0.5rem;
`;

export const ScaleFilterButton = styled.button`
  background-color: ${cssConstants.PRIMARY_LIGHT_BLUE};
  border: none;
  border-radius: 2px;
  color: ${cssConstants.PRIMARY_WHITE}
  font-weight: 300;
  height: 25px;
  min-width: 60px;
  padding: 0 1rem;
`;
