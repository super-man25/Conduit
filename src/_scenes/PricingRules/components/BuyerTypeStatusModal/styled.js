import styled from 'styled-components';
import { cssConstants } from '_constants';
import { H4, Flex, Input, Text } from '_components';
import { darken } from 'polished';

export const ModalOverlay = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  opacity: 0.4;
  background-color: ${cssConstants.PRIMARY_LIGHT_BLACK};
`;

export const ModalContent = styled.div`
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 500px;
  z-index: 9;
  box-shadow: 0 1px 10px rgba(0, 0, 0, 0.27);
`;

export const ModalHeader = styled.div`
  padding: 1.5rem;
  background-color: ${cssConstants.PRIMARY_LIGHTEST_GRAY};
  border-bottom: 1px solid ${cssConstants.PRIMARY_DARK_BLUE};
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
`;

export const ModalBody = styled.div`
  padding: 1rem 2rem;
  height: 400px;
  overflow-x: hidden;
  overflow-y: scroll;
  background-color: ${cssConstants.PRIMARY_LIGHTEST_GRAY};
  border-bottom: 1px solid ${cssConstants.PRIMARY_DARK_BLUE};
`;

export const ModalFooter = styled.div`
  padding: 1.5rem;
  background-color: ${cssConstants.PRIMARY_LIGHTEST_GRAY};
  border-bottom: ${cssConstants.PRIMARY_DARK_BLUE};
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 3px;
`;

export const BuyerTypeOption = Flex.extend`
  padding: 0 2rem;
  min-height: 45px;
  width: 100%;
  background-color: ${cssConstants.PRIMARY_LIGHTEST_GRAY};

  border-bottom: 1px solid ${cssConstants.PRIMARY_LIGHTER_GRAY};

  :hover {
    background-color: ${darken(0.05, cssConstants.PRIMARY_WHITE)};
  }
`;

export const FlexText = Text.extend`
  flex-basis: ${(props) => props.basis || '10%'};
`;

export const ErrorText = Text.extend`
  text-align: right;
`;

export const Title = H4.extend`
  margin: 0;
  margin-bottom: 2rem;
  color: ${cssConstants.PRIMARY_LIGHT_BLACK};
`;

export const FilterBuyerTypesInput = Input.extend`
  border-radius: 20px;
  height: 35px;
  width: 80px;
`;
