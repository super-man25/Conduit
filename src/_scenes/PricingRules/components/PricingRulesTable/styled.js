import styled from 'styled-components';
import { cssConstants } from '_constants';
import { PrimaryButton, Input, Box, SecondaryButton } from '_components';

export const EditPricingRuleInput = Input.extend`
  box-sizing: border-box;
  margin: 0 0.5rem 0 0;
  padding: 10px 8px;
  max-height: 35px;
  max-width: 80px;
`;

export const SavePricingRuleButton = PrimaryButton.extend`
  box-sizing: border-box;
  margin: 0;
  min-width: 50px;
  max-width: 75px;
  max-height: 35px;
`;

export const CancelEditingButton = SecondaryButton.extend`
  box-sizing: border-box;
  margin: 0;
  min-width: 50px;
  max-width: 75px;
  max-height: 35px;
  padding: 0 0.75rem;
`;

export const TableHeaderCell = styled.div`
  text-align: ${(props) => props.align || 'left'};
  color: ${cssConstants.PRIMARY_DARKEST_GRAY};
  font-weight: 500;
`;

export const PositionedBox = Box.extend`
  position: ${(props) => props.position};
  top: ${(props) => props.top};
  bottom: ${(props) => props.bottom};
  left: ${(props) => props.left};
  right: ${(props) => props.right};
  z-index: ${(props) => props.zIndex};
`;
export const Checkbox = styled.input.attrs({ type: 'checkbox' })`
  margin-right: 0.5rem;
`;
