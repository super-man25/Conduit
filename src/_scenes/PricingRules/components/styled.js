import styled, { css } from 'styled-components';

import { colors } from '_constants';
import { withClickAway } from '_hoc';
import { Input, Box } from '_components';

export const EditPricingRuleInput = styled(Input)`
  margin: 0 0.5rem 0 0;
  padding: 10px 8px;
  max-height: 35px;
  max-width: 80px;
`;

export const TableHeaderCell = styled.div`
  font-weight: bold;
`;

export const PositionedBox = styled(Box)`
  position: ${(props) => props.position};
  top: ${(props) => props.top};
  bottom: ${(props) => props.bottom};
  left: ${(props) => props.left};
  right: ${(props) => props.right};
  z-index: ${(props) => props.zIndex};
`;
export const Checkbox = styled.input.attrs(() => ({ type: 'checkbox' }))`
  margin-right: 0.5rem;
`;

export const MultiSelectContainer = withClickAway(styled.div`
  position: relative;
  cursor: pointer;
  display: flex;
  justify-content: space-between;
  align-items: center;
`);

export const MultiSelectMenu = styled.div`
  position: absolute;
  width: 250px;
  top: calc(100% + 8px);
  max-height: 350px;
  overflow-y: scroll;
  overflow-x: hidden;
  z-index: 10;
  border: 1px solid ${colors.lightGray};
  box-shadow: 0 20px 20px rgba(0, 0, 0, 0.06);

  transition: 0.1s ease-in-out all;
  opacity: 0;
  transform: translateY(20px);
  visibility: hidden;

  ${(props) =>
    props.show &&
    css`
      opacity: 1;
      transform: translateY(0);
      visibility: visible;
    `}
`;

export const MultiSelectOption = styled.div`
  padding: 12px 16px;
  background-color: ${(props) =>
    props.isActive ? colors.lightGray : colors.white};
  transition: 0.1s ease-in-out all;
  color: ${colors.black};
  white-space: normal;

  :not(:last-child) {
    border-bottom: 1px solid ${colors.lightGray};
  }

  :hover {
    cursor: ${(props) => (props.isActive ? 'default' : 'pointer')};
    background-color: ${colors.lightGray};
  }
`;

export const SplitButtonContainer = styled.div`
  flex-basis: 100%;
  display: flex;
  justify-content: center;
  margin-top: 15px;
`;

export const SplitButtonHalf = styled.span`
  padding: 8px;
  border: 1px solid ${colors.blue};
  display: inline-block;
  min-width: 35%;
  text-align: center;
  cursor: pointer;
  color: ${(props) => (props.isActive ? colors.white : colors.blue)};
  background-color: ${(props) => (props.isActive ? colors.blue : colors.white)};

  :not(:last-child) {
    border-bottom-left-radius: 2px;
    border-top-left-radius: 2px;
    border-right: none;
  }

  :last-child {
    border-bottom-right-radius: 2px;
    border-top-right-radius: 2px;
  }
`;

export const DropdownFilterInputContainer = styled.div`
  background-color: white;
  padding: 10px;
  border-bottom: 1px solid ${colors.lightGray};
`;

export const DropdownFilterInput = styled.input`
  width: 100%;
  padding: 10px 15px;
  border: 1px solid ${colors.lightGray};
  border-radius: 3px;
`;
