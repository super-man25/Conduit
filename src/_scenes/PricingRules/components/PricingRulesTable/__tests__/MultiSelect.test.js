import React from 'react';
import { shallow } from 'enzyme';
import { MultiSelect, GroupedSplitButton } from '../MultiSelect';
import { SplitButtonHalf } from '../styled';

describe('<MultiSelect />', () => {
  const props = {
    options: [
      { id: 2, name: 'Braves at Zephyrs' },
      { id: 5, name: 'Dogs at Cats' },
      { id: 17, name: 'Mariners at Zephyrs' }
    ],
    selected: [17],
    labelFn: jest.fn(),
    toggleSelectAll: jest.fn(),
    label: 'Events',
    isGroupable: false,
    updatePriceRuleProperty: jest.fn(),
    onItemClicked: jest.fn()
  };

  it('should render correctly', () => {
    const wrapper = shallow(<MultiSelect {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should set selectAllNext to false if all options selected', () => {
    const selected = [2, 5, 17];
    const wrapper = shallow(<MultiSelect {...props} selected={selected} />);

    expect(wrapper.state().selectAllNext).toBe(false);
  });

  it('should update the state if the grouped view is toggled', () => {
    const wrapper = shallow(<MultiSelect {...props} isGroupable={true} />);

    expect(wrapper.find(GroupedSplitButton).at(0).length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<GroupedSplitButton />', () => {
  const props = {
    isGrouped: true,
    selectGrouping: jest.fn()
  };

  it('should render correctly', () => {
    const wrapper = shallow(<GroupedSplitButton {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call selectGrouping prop function', () => {
    const fn = jest.fn();
    const wrapper = shallow(
      <GroupedSplitButton {...props} selectGrouping={fn} />
    );

    wrapper
      .find(SplitButtonHalf)
      .at(0)
      .simulate('click');

    expect(fn).toBeCalledWith(false);
  });
});
