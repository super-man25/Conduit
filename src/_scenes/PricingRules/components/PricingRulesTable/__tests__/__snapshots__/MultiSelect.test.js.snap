// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupedSplitButton /> should render correctly 1`] = `
<styled.div>
  <styled.span
    isActive={false}
    onClick={[Function]}
  >
    Uncategorized
  </styled.span>
  <styled.span
    isActive={true}
    onClick={[Function]}
  >
    By Category
  </styled.span>
</styled.div>
`;

exports[`<MultiSelect /> should render correctly 1`] = `
<withClickAway(styled.div)
  onClick={[Function]}
  onClickAway={[Function]}
>
  <Icon
    color="#070707"
    name="arrowDown"
    size={12}
  />
  <div
    style={
      Object {
        "position": "fixed",
        "zIndex": 9,
      }
    }
  >
    <styled.div
      show={false}
      style={
        Object {
          "maxHeight": Object {},
        }
      }
    >
      <styled.div>
        <styled.input
          onChange={[Function]}
          placeholder="Filter..."
          value=""
        />
      </styled.div>
      <Flex
        align="center"
        flexWrap="wrap"
        justify="space-between"
        padding="1rem"
        style={
          Object {
            "backgroundColor": "#FFFFFF",
          }
        }
      >
        <Styled(styled.p)
          size={13}
          weight={600}
        >
          Events
        </Styled(styled.p)>
        <Styled(styled.p)
          color="#38A9DB"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          Select All
        </Styled(styled.p)>
      </Flex>
      <MultiSelectView
        labelFn={[MockFunction]}
        onItemClicked={[MockFunction]}
        options={
          Array [
            Object {
              "id": 2,
              "name": "Braves at Zephyrs",
            },
            Object {
              "id": 5,
              "name": "Dogs at Cats",
            },
            Object {
              "id": 17,
              "name": "Mariners at Zephyrs",
            },
          ]
        }
        selected={
          Array [
            17,
          ]
        }
      />
    </styled.div>
  </div>
</withClickAway(styled.div)>
`;

exports[`<MultiSelect /> should update the state if the grouped view is toggled 1`] = `
<withClickAway(styled.div)
  onClick={[Function]}
  onClickAway={[Function]}
>
  <Icon
    color="#070707"
    name="arrowDown"
    size={12}
  />
  <div
    style={
      Object {
        "position": "fixed",
        "zIndex": 9,
      }
    }
  >
    <styled.div
      show={false}
      style={
        Object {
          "maxHeight": Object {},
        }
      }
    >
      <styled.div>
        <styled.input
          onChange={[Function]}
          placeholder="Filter..."
          value=""
        />
      </styled.div>
      <Flex
        align="center"
        flexWrap="wrap"
        justify="space-between"
        padding="1rem"
        style={
          Object {
            "backgroundColor": "#FFFFFF",
          }
        }
      >
        <Styled(styled.p)
          size={13}
          weight={600}
        >
          Events
        </Styled(styled.p)>
        <Styled(styled.p)
          color="#38A9DB"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          Select All
        </Styled(styled.p)>
        <GroupedSplitButton
          isGrouped={false}
          selectGrouping={[Function]}
        />
      </Flex>
      <MultiSelectView
        labelFn={[MockFunction]}
        onItemClicked={[MockFunction]}
        options={
          Array [
            Object {
              "id": 2,
              "name": "Braves at Zephyrs",
            },
            Object {
              "id": 5,
              "name": "Dogs at Cats",
            },
            Object {
              "id": 17,
              "name": "Mariners at Zephyrs",
            },
          ]
        }
        selected={
          Array [
            17,
          ]
        }
      />
    </styled.div>
  </div>
</withClickAway(styled.div)>
`;
